modules/account-lib/src/coin/dot/transaction.ts

```diff
@@ -346,10 +348,30 @@ export class Transaction extends BaseTransaction {
     }
   }
 
+  constructSignedPayload(signature: Buffer): void {
+    const edSignature = `0x00${signature.toString('hex')}` as HexString;
+
+    this._signedTransaction = construct.signedTx(this._dotTransaction, edSignature, {
+      registry: this._registry,
+      metadataRpc: this._dotTransaction.metadataRpc,
+    });
+
+    const recoveredSignature = utils.recoverSignatureFromRawTx(this._signedTransaction, { registry: this._registry });
```

Comment:
Is this the same as `${signature.toString('hex')}`?

Reply:
It should be, but I figured that the best way to extract the signature is from the signed transaction itself rather than simply adding the signature provided from addSignature() to the list of signatures. Ideally it should  always be the same but though i should get it from the best source of truth and figured it would be from the signed transaction itself.

let me know what you think.

Reply:
Hmm if we are confident that these two will always have the same value, it feels to me like duplicated computation here (since looks like`recoveredSignature` is expected to be the same as `edSignature`) But if we are unsure, maybe we can run an assert and see if we hit situations that they they are different, and deprecate the recoverySignature in the future once we are certain. Like:
```ts
assert(recoveredSignature === edSignature, new Error("derived signature is different from provided signature");
this._signature = signature.toString('hex');
```
or something

Reply:
or we can run this case in unit tests

Reply:
Actually it's not expected to be the same as edSignature since edSignature prepends 0x00. 

The output of recoveredSignature = signature.toString('hex'). Test pass if I do 

this._signature = signature.toString('hex');

Im going to remove the double computation and just do the above instead.